File: taller/AMarotoChivite-taller/AMarotoChivite-taller-Java/src/main/java/simulacionTaller/factories/PlaterFactory.java
package simulacionTaller.factories;


import simulacionTaller.models.Plater;
import simulacionTaller.models.Worker;

import static simulacionTaller.utils.randomInfo.*;

public class PlaterFactory {
    public static Plater create() {
        String name = randomName();
        int experienceYear = randomAnno();
        int salary = randomSalary();
        int dailyHours = randomDailyHours();

        return new Plater(name, experienceYear, salary, dailyHours, Worker.BossAssigment.WITHOUT_BOSS);
    }
}

File: taller/AMarotoChivite-taller/AMarotoChivite-taller-kotlin/src/main/kotlin/simulacionTaller/factories/PlaterFactory.kt
package simulacionTaller.factories

import simulacionTaller.models.Plater
import simulacionTaller.models.Worker.BossAssigment
import simulacionTaller.utils.randomAnno
import simulacionTaller.utils.randomDailyHours
import simulacionTaller.utils.randomName
import simulacionTaller.utils.randomSalary

object PlaterFactory {
    @JvmStatic
    fun create(): Plater {
        val name = randomName()
        val experienceYear = randomAnno()
        val salary = randomSalary()
        val dailyHours = randomDailyHours()
        return Plater(name, experienceYear, salary.toDouble(), dailyHours, BossAssigment.WITHOUT_BOSS)
    }
}